src/cluster.py

```diff
@@ -1057,11 +1058,16 @@ def _synchronous_node_count(self) -> int:
     def update_synchronous_node_count(self) -> None:
         """Update synchronous_node_count to the minority of the planned cluster."""
         # Try to update synchronous_node_count.
+        member_units = json.loads(self.charm.app_peer_data.get("members_ips", "[]"))
```

Comment:
Check the size of the members list in the peer data. This is necessary to make sure that the cluster successfully botstrapped.

Reply:
I would go with a clean flag but  your approach looks working.